config/initializers/content_security_policy.rb

```diff
@@ -2,17 +2,34 @@
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
-# Rails.application.config.content_security_policy do |p|
-#   p.default_src :self, :https
-#   p.font_src    :self, :https, :data
-#   p.img_src     :self, :https, :data
-#   p.object_src  :none
-#   p.script_src  :self, :https
-#   p.style_src   :self, :https, :unsafe_inline
-#
-#   # Specify URI for violation reports
-#   # p.report_uri "/csp-violation-report-endpoint"
-# end
+if Rails.env.production?
+  assets_host = Rails.configuration.action_controller.asset_host || "https://#{ENV['WEB_DOMAIN'] || ENV['LOCAL_DOMAIN']}"
+  data_hosts = [assets_host]
+
+  if ENV['S3_ENABLED'] == 'true'
+    attachments_host = ENV['S3_ALIAS_HOST'] || ENV['S3_CLOUDFRONT_HOST'] || ENV['S3_HOSTNAME'] || "s3-#{ENV['S3_REGION'] || 'us-east-1'}.amazonaws.com"
+  elsif ENV['SWIFT_ENABLED'] == 'true'
+    attachments_host = ENV['SWIFT_OBJECT_URL']
+  else
+    attachments_host = nil
+  end
+  data_hosts << attachments_host unless attachments_host.nil?
+
+  Rails.application.config.content_security_policy do |p|
+    p.base_uri        :none
+    p.default_src     :none
+    p.frame_ancestors :none
```

Comment:
I think that frame_ancestors is necessary for the embed of toot.

On my server I set "frame-ancestors https:" and running the Mastodon.

Reply:
The controller responsible for handling embedded toots overwrites this!

Reply:
I'm sorry, I did not know that.